app/controllers/pages_controller.rb

```diff
@@ -1,4 +1,6 @@
 class PagesController < ApplicationController
   def index
+    return if !params[:city_name] || Location.all.any? { |a| a.city_name == params[:city_name] }
```

Comment:
With rails ActiveRecord, you have to be mindful when you potentially evaluate SQL queries with Enumerable methods: https://ruby-doc.org/core-2.5.1/Enumerable.html

If you're not sure, you can test it out in rails console first:
```
$ rails c 
irb(main):001:0> Location.all.any? { |a| a.city_name == "Providence" }
  Location Load (0.4ms)  SELECT "locations".* FROM "locations"
=> true
```

You can see above that we query all records from the locations table which means they are all loaded into memory, and then enumerated for the `city_name ==` condition. What want to do is only query what we need and nothing more with the ActiveRecord query methods:
https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Relation.html#method-i-any-3F
https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Relation.html#method-i-empty-3F
```
irb(main):002:0> Location.where(city_name: "Providence").any?
  Location Exists (0.7ms)  SELECT  1 AS one FROM "locations" WHERE "locations"."city_name" = ? LIMIT ?  [["city_name", "Providence"], ["LIMIT", 1]]
=> true
irb(main):003:0> Location.where(city_name: "Providence").empty?
  Location Exists (0.5ms)  SELECT  1 AS one FROM "locations" WHERE "locations"."city_name" = ? LIMIT ?  [["city_name", "Providence"], ["LIMIT", 1]]
=> false
```
Now we're only returning a single scalar from the database if there are rows and ActiveRecord checks the row count to give us the appropriate boolean result.